### PR TITLE
[12.0][MIG] partner_coc

### PR DESCRIPTION
**Pull request by tarteo**
Depends on:
https://github.com/OCA/partner-contact/pull/644

### CODE REVIEW

**Review by CasVissers-360ERP — APPROVED** [review]
LGTM, functional test. Hate it though that this module depends on partner_identification. Most companies just want the possibility to register the CoC.

**Review by astirpe — APPROVED** [review]
I agree with @CasVissers that the dependency on `partner_identification` is not necessary and forces the users to download and install two modules, just for adding one field.
Anyway I'm not blocking this PR since it's just the pure porting of the module.
Code review and tests: OK.

**Comment by NL66278** [comment]
I think is is quite usefull to have a single framework for controlling all kinds of ID's and related documents that have a limited validity. From driving licences, to passports, or code of conduct declarations.

File: partner_coc/models/res_partner.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    coc_registration_number = fields.Char(
+        string='CoC Registration Number',
+        compute=lambda s: s._compute_identification(
+            'coc_registration_number', 'coc',
+        ),
+        inverse=lambda s: s._inverse_identification(
+            'coc_registration_number', 'coc',
+        ),
+        search=lambda s, *a: s._search_identification(
+            'coc_registration_number', 'coc', *a
```
Comment (NL66278) [inline partner_coc/models/res_partner.py]: Please cherry-pick this commit fro 11.0 or search will not work: https://github.com/OCA/partner-contact/commit/a524277006433a72c978fd45ed5daf87a1f38f05

**Review by NL66278 — CHANGES_REQUESTED** [review]
Correction needed in search method.

File: partner_coc/models/res_partner.py (hunk 1 above)
Comment (tarteo) [reply partner_coc/models/res_partner.py]: Thanks cherry picked it :+1:

**Review by NL66278 — APPROVED** [review]

**Pull request merged**